makefiles/cflags.inc.mk

```diff
@@ -25,28 +25,27 @@ ifeq ($(shell $(CC) -fno-delete-null-pointer-checks -E - 2>/dev/null >/dev/null
   endif
 endif
 
+# Template for testing a compiler flag and adding it to CFLAGS (errors usually
+# happens when using older toolchains which do not support the given flags)
+define cflags_test_and_add
+  ifeq ($(shell $(CC) $(1) -E - 2>/dev/null >/dev/null </dev/null ; echo $$?),0)
+    CFLAGS += $(1)
+  endif
+endef
 # Use colored compiler output if the compiler supports this and if this is not
 # disabled by the user
 ifeq ($(CC_NOCOLOR),0)
-  ifeq ($(shell $(CC) -fdiagnostics-color -E - 2>/dev/null >/dev/null </dev/null ; echo $$?),0)
-    CFLAGS += -fdiagnostics-color
-  endif
+  $(eval $(call cflags_test_and_add,-fdiagnostics-color))
 endif
 
 # Fast-out on old style function definitions.
 # They cause unreadable error compiler errors on missing semicolons.
 # Worse yet they hide errors by accepting wildcard argument types.
-ifeq ($(shell $(CC) -Wstrict-prototypes -Werror=strict-prototypes -Wold-style-definition -Werror=old-style-definition -E - 2>/dev/null >/dev/null </dev/null ; echo $$?),0)
-  # duplicated parameters don't hurt
-  CFLAGS += -Wstrict-prototypes -Wold-style-definition
-  CXXUWFLAGS += -Wstrict-prototypes -Wold-style-definition
```

Comment:
Just for clarity, I would prefer moving the `CXXUWFLAGS` and removing the useless `WERROR` case to a separate "cleanup" commit.